core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -55,7 +56,8 @@
  *
  * @author Daniel Kayiwa, Drew Roos
  */
-public class FormDef implements IFormElement, Localizable, Persistable, IMetaData {
+public class FormDef extends ActionController
```

Comment:
It seems a bit unwieldy for me to require action triggering elements to extend the controller by their nature. the class already has to implement the processor, it seems like it would be cleaner to me to have an interface which exposed access to a controller object by composition. IE: The formdef has an "ActionController" object that handles serialization and the events/dispatch, and you call formDef.getActionController().registerEventListener() rather than making it "intrinsic" to the class itself.